lib/deep-speech-interface.js

```diff
@@ -84,67 +84,77 @@ class DeepSpeechInterface {
-    const localLMBinary = path.join(this._modelsDir, 'local_lm.binary');
-    fs.writeFileSync(localLMTxt, finalGrammar.join('\n'));
+    const lmText = path.join(this._modelsDir, 'lm.txt');
+    fs.writeFileSync(lmText, finalGrammar.join('\n'));
 
-    const lmplz = spawnSync(
-      path.join(this._binDir, 'lmplz'),
+    const generateLm = spawnSync(
+      'python3',
       [
-        '--memory', '64M',
-        '--order', '2',
+        path.join(this._binDir, 'generate_lm.py'),
+        '--input_txt', lmText,
+        '--output_dir', this._modelsDir,
+        '--top_k', '500000',
+        '--kenlm_bins', this._binDir,
+        '--arpa_order', '5',
+        '--max_arpa_memory', '50%',
+        '--arpa_prune', '0|0|1',
+        '--binary_a_bits', '255',
+        '--binary_q_bits', '8',
+        '--binary_type', 'trie',
         '--discount_fallback',
-        '--text', localLMTxt,
-        '--arpa', localLMArpa,
       ]
     );
 
     if (this._debug) {
-      console.debug('lmplz: status:', lmplz.status);
-      console.debug('lmplz: stdout:', lmplz.stdout.toString());
-      console.debug('lmplz: stderr:', lmplz.stderr.toString());
+      console.debug('generate_lm.py: status:', generateLm.status);
+      console.debug('generate_lm.py: stdout:', generateLm.stdout.toString());
+      console.debug('generate_lm.py: stderr:', generateLm.stderr.toString());
     }
 
-    const buildBinary = spawnSync(
-      path.join(this._binDir, 'build_binary'),
+    const generateScorer = spawnSync(
+      path.join(this._binDir, 'generate_scorer_package'),
       [
-        '-a', '255',
-        '-q', '8',
-        'trie',
-        localLMArpa,
-        localLMBinary,
+        '--lm', path.join(this._modelsDir, 'lm.binary'),
+        '--vocab', lmText,
+        '--package', path.join(this._modelsDir, 'kenlm.scorer'),
+        '--default_alpha', '0.931289039105002',
+        '--default_beta', '1.1834137581510284',
+        '--force_utf8', 'true',
+        // https://github.com/mozilla/DeepSpeech/issues/3205
+        '--alphabet', 'fake',
       ]
     );
 
     if (this._debug) {
-      console.debug('build_binary: status:', buildBinary.status);
-      console.debug('build_binary: stdout:', buildBinary.stdout.toString());
-      console.debug('build_binary: stderr:', buildBinary.stderr.toString());
+      console.debug(
+        'generate_scorer_package: status:',
+        generateScorer.status
+      );
+      console.debug(
+        'generate_scorer_package: stdout:',
+        generateScorer.stdout.toString()
+      );
+      console.debug(
+        'generate_scorer_package: stderr:',
+        generateScorer.stderr.toString()
+      );
     }
 
     if (this._debug) {
       console.debug(`Loading model from ${this._modelsDir}`);
     }
 
-    const modelJson =
-      JSON.parse(fs.readFileSync(path.join(this._assetsDir, 'info.json')));
-
     const extension = process.arch === 'x64' ? 'pbmm' : 'tflite';
```

Comment:
Why dont you just rely always on tflite ?

Reply:
It seemed like `pbmm` was preferred on AMD64, so I just went with that. I either had to swap model files or deepspeech Node packages.

Reply:
No, it's more for historic reasons, there was only tensorflow runtime, we added tflite after, and it was made default on some packages because tensorflow runtime was not a serious solution (10x slower)

Reply:
Ok, always using tflite.